src/UnisonLocal/Page/HomePage.elm

```diff
@@ -1,42 +1,208 @@
 
+import Code.BranchRef as BranchRef exposing (BranchSlug(..))
+import Code.Perspective as Perspective
+import Dict exposing (Dict)
+import Html exposing (Html, div, h2, p, text)
+import Json.Decode as Decode
+import Lib.HttpApi as HttpApi
+import Lib.Util as Util
+import RemoteData exposing (RemoteData(..), WebData)
+import UI.Click as Click
 import UI.PageContent as PageContent
 import UI.PageLayout as PageLayout exposing (PageFooter(..))
-import UI.StatusBanner as StatusBanner
+import UI.PageTitle as PageTitle
+import UI.Tag as Tag
+import UnisonLocal.Api as LocalApi
 import UnisonLocal.AppContext exposing (AppContext)
 import UnisonLocal.AppDocument as AppDocument exposing (AppDocument)
 import UnisonLocal.AppHeader as AppHeader
+import UnisonLocal.ProjectName as ProjectName exposing (ProjectName)
+import UnisonLocal.Route as Route
+
+
+
+-- MODEL
 
 
 type alias Model =
-    ()
+    { projects : Projects }
+
+
+type alias Projects =
+    -- Since a `Dict` requires a key of type `comparable`
+    -- `ProjectName` is made available in the value
+    -- for further processing
+    Dict String ( ProjectName, List BranchSlug )
 
 
 init : AppContext -> ( Model, Cmd Msg )
-init _ =
-    ( (), Cmd.none )
+init appContext =
+    ( { projects = Dict.empty }
+    , fetchProjects FetchProjectsFinished
+        |> HttpApi.perform appContext.api
+    )
+
+
+
+-- UPDATE
 
 
 type Msg
-    = NoOp
+    = FetchProjectsFinished (WebData (List ProjectName))
+    | FetchProjectBranchesFinished (WebData ( ProjectName, List BranchSlug ))
 
 
 update : AppContext -> Msg -> Model -> ( Model, Cmd Msg )
-update _ _ model =
-    ( model, Cmd.none )
+update appContext msg model =
+    case msg of
+        FetchProjectsFinished (Success projectNames) ->
+            ( { projects =
+                    projectNames
+                        |> List.map
+                            (\p ->
+                                ( ProjectName.toString p
+                                , ( p, [] )
+                                )
+                            )
+                        |> Dict.fromList
+              }
+            , projectNames
+                |> List.map
+                    (fetchProjectBranches FetchProjectBranchesFinished
+                        >> HttpApi.perform appContext.api
+                    )
+                |> Cmd.batch
+            )
+
+        FetchProjectBranchesFinished (Success ( projectName, branches )) ->
+            ( { model
+                | projects =
+                    model.projects
+                        |> Dict.insert
+                            (ProjectName.toString projectName)
+                            ( projectName, branches )
+              }
+            , Cmd.none
+            )
+
+        _ ->
+            ( model, Cmd.none )
+
+
+
+-- EFFECTS
+
+
+fetchProjects :
+    (WebData (List ProjectName) -> msg)
+    -> HttpApi.ApiRequest (List ProjectName) msg
+fetchProjects finishedMsg =
```

Comment:
I don't feel strongly, but this function and `fetchProjectBranches` are both fine to use `FetchProjectsFinished ` and `FetchProjectBranchesFinished` instead of the argument.

Reply:
Good suggestion, I simplified this